app/repositories/application/GeneralApplicationRepository.scala

```diff
@@ -364,15 +365,30 @@ class GeneralApplicationMongoRepository(timeZoneService: TimeZoneService,
       BSONDocument("applicationStatus" -> BSONDocument("$ne" -> WITHDRAWN))
     )))
 
-  override def findFailedTestForNotification(failedTestType: FailedTestType): Future[Option[NotificationFailedTest]] = {
-    val query = BSONDocument("$and" -> BSONArray(
-      BSONDocument("applicationStatus" -> failedTestType.appStatus),
-      BSONDocument(s"progress-status.${failedTestType.notificationProgress}" -> BSONDocument("$ne" -> true)),
-      BSONDocument(s"progress-status.${failedTestType.receiveStatus}" -> true)
-    ))
+  override def findTestForNotification(notificationType: NotificationTestType): Future[Option[TestResultNotification]] = {
+    val query = Try{ notificationType match {
```

Comment:
At the present time we have just two types so you'll never end up in the unknown case, but this check is from when (and if) we'll add more of them. In that scenario you won't get any compile error but just a warning that the match is not exhaustive and then a runtime exception because of the unmatched case. Try is there just to wrap that exception in a Future failed rather than propagate it. Let me know if that is ok with you.